lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2383,6 +2385,42 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
     if (VD->getAttrs().hasAttribute<DynamicReplacementAttr>())
       TC.checkDynamicReplacementAttribute(VD);
   }
+    
+  void checkForEmptyOptionSet(VarDecl *VD) {
```

Comment:
I might suggest making `VD` `const` just to make it clear that it's not going to modify the AST to fix any issues.